checker/src/call_visitor.rs

```diff
@@ -1007,7 +1007,20 @@ impl<'call, 'block, 'analysis, 'compilation, 'tcx, E>
         precondition!(self.actual_args.len() == 1);
 
         if let Some(tag) = self.extract_tag_kind_and_propagation_set() {
-            let source_path = Path::new_deref(self.actual_args[0].0.clone())
+            let source_pointer_path = self.actual_args[0].0.clone();
+            let source_pointer_rustc_type = self
+                .block_visitor
+                .bv
+                .type_visitor
+                .get_path_rustc_type(&source_pointer_path, self.block_visitor.bv.current_span);
```

Comment:
Instead of doing this expensive and brittle operation, you can just get the type from self.actual_argument_types.